apps/condo/domains/ticket/schema/TicketComment.test.js

```diff
@@ -15,370 +15,940 @@ const { makeClientWithProperty } = require('@condo/domains/property/utils/testSc
+                const employeeClient = await makeClientWithNewRegisteredAndLoggedInUser()
+                const residentClient = await makeClientWithResidentUser()
+
+                const unitName = faker.random.alphaNumeric(5)
+                const content1 = faker.lorem.sentence()
+                const content2 = faker.lorem.sentence()
+                const { phone } = residentClient.userAttrs
+
+                const [organization] = await createTestOrganization(adminClient)
+                const [property] = await createTestProperty(adminClient, organization)
+                const [role] = await createTestOrganizationEmployeeRole(adminClient, organization, {
+                    canManageTickets: true,
+                    canManageTicketComments: true,
+                })
+                await createTestOrganizationEmployee(adminClient, organization, employeeClient.user, role)
+                await createTestResident(adminClient, residentClient.user, organization, property, {
+                    unitName,
+                })
+                const [contact] = await createTestContact(adminClient, organization, property, {
+                    phone,
+                    unitName,
+                })
+                const [ticket] = await createTestTicket(employeeClient, organization, property, {
+                    unitName,
+                    contact: { connect: { id: contact.id } },
+                    canReadByResident: true,
+                })
+
+                const [commentFromResident] = await createTestTicketComment(residentClient, ticket, residentClient.user, {
+                    type: COMMENT_TYPE.RESIDENT,
+                    content: content1,
+                })
+                const [commentFromEmployee] = await createTestTicketComment(employeeClient, ticket, employeeClient.user, {
+                    type: COMMENT_TYPE.RESIDENT,
+                    content: content2,
+                })
+
+                const { data: { objs: comments } } = await TicketComment.getAll(residentClient, {}, { sortBy: 'createdAt_ASC', raw: true })
+
+                expect(comments).toHaveLength(2)
+                expect(comments[0].id).toEqual(commentFromResident.id)
+                expect(comments[0].content).toEqual(content1)
+                expect(comments[1].id).toEqual(commentFromEmployee.id)
+                expect(comments[1].content).toEqual(content2)
             })
-            const [ticket] = await createTestTicket(admin, organizationFrom, propertyFrom)
-            const [comment] = await createTestTicketComment(admin, ticket, clientFrom.user)
-            const payload = {
-                content: faker.random.alphaNumeric(10),
-            }
 
-            await expectToThrowAccessDeniedErrorToObj(async () => {
-                await updateTestTicketComment(clientTo, comment.id, payload)
+            it('cannot read comments in not his resident ticket', async () => {
```

Comment:
We can also check the read after Resident.deletedAt. And read after Property.deletedAt and Ticket.deletedAt